crates/ol/mempool/src/events.rs

```diff
@@ -0,0 +1,135 @@
+//! Event types for mempool notifications.
+//!
+//! Provides types for subscribing to mempool state changes such as transaction additions,
+//! removals, and evictions.
+
+use strata_identifiers::OLTxId;
+
+/// Events emitted by the mempool for state changes.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum MempoolEvent {
+    /// A transaction was successfully added to the mempool.
+    TransactionAdded {
+        /// Transaction ID.
+        txid: OLTxId,
+        /// Priority score assigned to the transaction.
+        priority: u64,
+    },
+
+    /// A transaction was removed from the mempool.
+    TransactionRemoved {
+        /// Transaction ID.
+        txid: OLTxId,
+        /// Reason for removal.
+        reason: RemovalReason,
+    },
+
+    /// A transaction was evicted due to capacity limits.
+    TransactionEvicted {
+        /// Transaction ID that was evicted.
+        txid: OLTxId,
+    },
```

Comment:
Make all of these struct variants their own structs.  It's impossible to pass a specific variant's to a function (in Rust), but it's an extremely common pattern where you want to have a function that matches on an event type and then dispatches to a handler function with that variant's data.